# Parcial 1 · Machine Learning (Primer Corte)
**Formato:** Notebook de Python (Colab/Jupyter)

**Instrucciones**
- Responde directamente en las celdas indicadas con *Tu respuesta aquí*.
- No borres el enunciado de las preguntas.
- No uses gráficos; solo filtros, agrupaciones y cálculos básicos.
- Mantén el código y comentarios en **español**.
- **Entrega**: sube este `.ipynb` resuelto a la plataforma indicada.

**Dataset esperado (`students.csv`)**
Columnas: `gender, race/ethnicity, parental level of education, lunch, test preparation course, math score, reading score, writing score`.

**Estructura del parcial**
- **Sección A — Fundamentos (6 preguntas)**
- **Sección B — EDA con filtros y agrupaciones (6 preguntas)**

**Puntaje**: 100 puntos (≈ 8.3 por pregunta).

---

In [1]:
!wget https://raw.githubusercontent.com/javierherrera1996/lecture_analytics/main/StudentsPerformance.csv

--2025-09-17 20:47:28--  https://raw.githubusercontent.com/javierherrera1996/lecture_analytics/main/StudentsPerformance.csv
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 72036 (70K) [text/plain]
Saving to: ‘StudentsPerformance.csv’

StudentsPerformance 100%[===================>]  70.35K  --.-KB/s    in 0.02s   

2025-09-17 20:47:28 (3.11 MB/s) - ‘StudentsPerformance.csv’ saved [72036/72036]



### Dependencias (ejecuta esta celda primero)

In [2]:
import pandas as pd
import numpy as np

## Carga del dataset
Sube un archivo **`students.csv`** (mismo directorio del notebook) o ajusta la ruta en la celda. Si no tienes el archivo, puedes **generar una muestra mínima** con la celda opcional.


In [3]:
df = pd.read_csv("StudentsPerformance.csv")

#### (Opcional) Generar una muestra mínima si no tienes el CSV

## Sección A — Fundamentos de Machine Learning (Conceptual, sin métricas ni gráficos)

### A1. Supervisado vs No Supervisado
**Describe** con tus palabras la diferencia entre **aprendizaje supervisado** y **no supervisado**. Da **1 ejemplo** de cada uno aplicado a educación o negocios.

_Tu respuesta aquí (Markdown):_

**Aprendizaje Supervisado:** Se le entregan unos datos ya “Etiquetados” y la maquina va a aprender de estos patrones. Se le da un Feedback, si esta bien o no. Ejemplo: Un banco usa datos de clientes pasados (quién pagó y quién no pagó un préstamo) para entrenar un modelo que prediga si un nuevo cliente pagará o no.

**Aprendizaje No Supervisado:** Se le entregan los datos, pero ya no tiene el feedback, sino que la maquina aprende por medio de patrones teniendo en cuenta las caracteristicas que se le entregaron. Ejemplo: una tienda deja que un programa agrupe solo a sus clientes. Descubre que hay grupos distintos: unos compran barato, otros prefieren lo más caro, y otros solo en promociones, asi pueden ir definiendo que les gusta a sus clientes



### A2. Variable dependiente e independientes
Si queremos predecir `math score`, ¿cuál es la **variable dependiente**? Menciona **dos** posibles **variables independientes** justificando por qué.

_Tu respuesta aquí (Markdown):_

Debido a que el resultado que queremos explicar o predecir es Math Score esa serua la variable dependiente

**Variables Independientes:** El test de preparacion para los examenes, ya que hacer o no un curso de preparación puede influir directamente en el desempeño académico. y El nivel de estudio de los padres ya que si estos tienen niveles mas altos puden ayudar al hijo


### A3. Clasificación binaria
Si clasificamos estudiantes en **Aprobado** (≥ 60) vs **No aprobado**, ¿qué tipo de aprendizaje es? Justifica brevemente.

_Tu respuesta aquí (Markdown):_

Seria aprendizaje supervisado ya que se le estarian entregando las etiquetas al programa y como identificarlas. El programa aprenderia a predecir en que grupo caeran los estudiantes

### A4. Casos de negocio (supervisado)
Da **dos** ejemplos de problemas de negocio donde usarías aprendizaje **supervisado** y explica por qué.

_Tu respuesta aquí (Markdown):_

1. Predescir si un cliente pagaria un prestamo o no, ya que teneindo en cuenta los datos previos de otros clientes, el modelo aprende a reconocer los patrones y predecir si pagaran o no

2. Predecir si un producto nuevo trendra buena o mala reseña, ya que el modelo se peude entrenar con reseñas anterioress que esten clasificadas como buenas o malas

### A5. Casos de negocio (no supervisado)
Da **dos** ejemplos de problemas de negocio donde usarías aprendizaje **no supervisado** y explica por qué.

_Tu respuesta aquí (Markdown):_

1. Segmentacion de clientes en una tienda, es no supervisado ya que aunque tenemos datos de los clientes no los tenemos agrupados o no sabemos a que grupo pertenecen asi que el modelo encuentra patrones y los va agrupando

2. La deteccion de anomalias en transacciones bancarias ya que se tiene la informacion de las transacciones pero no la clseificacion de Buena o mala, asi que el modelo aprede a identificar los movimientos extraños que podrian ser fraude

### A6. Segmentación por similitud
Si quieres segmentar estudiantes por similitud de `reading score` y `writing score`, ¿qué tipo de aprendizaje usarías y por qué?

_Tu respuesta aquí (Markdown):_

Seria aprendizaje no supervisado ya que se tienen los datos de los estudiantes pero no estan clasificados y el modelo los agruparia segun su similitud de puntajes

## Sección B — EDA (solo filtros y agrupaciones, foco de negocio)

### B7. Promedio de matemáticas por género
Calcula el **promedio de `math score` por `gender`**.
- ¿Qué género tiene mejor promedio?
- **Conclusión de negocio**: en 1–2 líneas, ¿qué harías con esta información?

_Escribe tu código y comentarios:_

In [21]:
promedios = df.groupby("gender")["math score"].mean().sort_values(ascending=False)
print(promedios)

gender
male      68.728216
female    63.633205
Name: math score, dtype: float64


Los hombres tienen mejor promedio. Con esta informacion se podria hacer un programa de refuerzo para las mujeres y que puedan subir y equilibrar el promedio con los hombres

### B8. Lectura por grupo étnico
Calcula el **promedio de `reading score` por `race/ethnicity`**.
- ¿Qué grupo tiene mejor desempeño?
- **Aplicación**: ¿cómo usarías esto en un programa de apoyo escolar?

_Escribe tu código y comentarios:_

In [20]:
Pomedios = df.groupby("race/ethnicity")["reading score"].mean().sort_values(ascending=False)
print(Pomedios)


race/ethnicity
group E    73.028571
group D    70.030534
group C    69.103448
group B    67.352632
group A    64.674157
Name: reading score, dtype: float64


El gupo que mejor desempeño tuvo fue el Grupo E con 73.028571

Teniendo en cuenta estos resultados podria ponerle clases de apoyo en lectura a los grupos que obtuvieron menor puntaje como el A y el B. Buscando que mejoren su promedio y alcancen a los grupos con promedios mas altos

### B9. Impacto del tipo de almuerzo
Filtra estudiantes con `lunch = free/reduced` y calcula el **promedio de `math score` y `reading score`**. Compáralo con `lunch = standard`.
- **Conclusión de negocio**: ¿qué acción propones?

_Escribe tu código y comentarios:_

In [28]:
df.groupby("lunch")[["math score", "reading score"]].mean().sort_values(by="math score", ascending=False)

math score  reading score
lunch                                  
standard       70.034109      71.654264
free/reduced   58.921127      64.653521

Se podría implementar un programa de mejor calidad con alimentos más nutritivos para quienes reciben lunch free/reduced, ya que la calidad de la dieta es un factor que influye en el promedio de los estudiantes

### B10. Educación parental y rendimiento
Agrupa por `parental level of education` y calcula el **promedio de `math score`**.
- ¿Qué nivel se asocia con mejores resultados?
- **Acción**: ¿qué intervención o programa harías basado en esto?

_Escribe tu código y comentarios:_

In [24]:
df.groupby("parental level of education")["math score"].mean().sort_values(ascending=False)

parental level of education
master's degree       69.745763
bachelor's degree     69.389831
associate's degree    67.882883
some college          67.128319
some high school      63.497207
high school           62.137755
Name: math score, dtype: float64

Se podria implementar un programa de apoyo extra en matemáticas para estudiantes cuyos padres tienen menor nivel educativo, por ejemplo tutorías gratuitas o talleres de refuerzo, para reducir la brecha de rendimiento

### B11. Curso de preparación y puntaje
Filtra estudiantes que **completaron** `test preparation course` y calcula el **promedio de `math score`**. Compáralo con quienes **no** lo completaron.
- **Decisión**: ¿financiarías este curso? Explica.

_Escribe tu código y comentarios:_

In [29]:
df.groupby("test preparation course")["math score"].mean().sort_values(ascending=False)

test preparation course
completed    69.695531
none         64.077882
Name: math score, dtype: float64

Si seguiria financiando el curso ya que aunque la brecha es pequeña si se ve una mejora en el promedio de los estudiantes y pondria mas bien como objetivo que el mayor numero de estudiantes lo termine para obtener un promedio general mucho mas alto

### B12. Doble agrupación (negocio)
Crea una **tabla** agrupada por `gender` y `test preparation course` con el **promedio de `writing score`**.
- ¿Qué combinación muestra mejores resultados?
- **Recomendación**: ¿a quién priorizas para una beca del curso?

_Escribe tu código y comentarios:_

In [30]:
df.groupby(["gender", "test preparation course"])["writing score"].mean().sort_values(ascending=False)

gender  test preparation course
female  completed                  78.793478
male    completed                  69.793103
female  none                       68.982036
male    none                       59.649351
Name: writing score, dtype: float64

Teniendo en cuenta estos resultados priorizaria una beca para las mujeres que hayan completado el test ya que son las que presentan un promedio de escritura mas alto